imap/monitor.go

```diff
@@ -147,41 +150,46 @@ func checkForNewEmails(im models.IMAP) {
 				}
 			}
 
-			body := string(append(m.Email.Text, m.Email.HTML...)) // Not sure if we need to check the Text as well as the HTML. Perhaps sometimes Text only emails won't have an HTML component?
-			rid := goPhishRegex.FindString(body)
+			rids, err := checkRIDs(m.Email) // Search email Text, HTML, and each attachment for rid parameters
 
-			if rid != "" {
-				rid = rid[5:]
-				log.Infof("User '%s' reported email with rid %s", m.Email.From, rid)
-				result, err := models.GetResult(rid)
-				if err != nil {
-					log.Error("Error reporting GoPhish email with rid ", rid, ": ", err.Error())
-					reportingFailed = append(reportingFailed, m.UID)
-				} else {
-					err = result.HandleEmailReport(models.EventDetails{})
+			if err != nil {
+				log.Errorf("Error searching email for rids from user '%s': %s", m.Email.From, err.Error())
+			} else {
+				if len(rids) < 1 {
+					// In the future this should be an alert in Gophish
+					log.Infof("User '%s' reported email with subject '%s'. This is not a GoPhish campaign; you should investigate it.", m.Email.From, m.Email.Subject)
+				}
+				for rid := range rids {
+					log.Infof("User '%s' reported email with rid %s", m.Email.From, rid)
+					result, err := models.GetResult(rid)
 					if err != nil {
-						log.Error("Error updating GoPhish email with rid ", rid, ": ", err.Error())
+						log.Error("Error reporting GoPhish email with rid ", rid, ": ", err.Error())
+						reportingFailed = append(reportingFailed, m.SeqNum)
 					} else {
-						if im.DeleteReportedCampaignEmail == true {
-							campaignEmails = append(campaignEmails, m.UID)
+						err = result.HandleEmailReport(models.EventDetails{})
+						if err != nil {
+							log.Error("Error updating GoPhish email with rid ", rid, ": ", err.Error())
+						} else {
+							if im.DeleteReportedCampaignEmail == true {
+								campaignEmails = append(campaignEmails, m.SeqNum)
```

Comment:
You might consider renaming this to `deleteEmails` or some other name that indicates the emails are scheduled to be deleted. Right now, that's the only reason we have this list.

By doing that, you can drop the `im.DeleteReportedCampaignEmail == true` check later on, since you're already checking it here.

That is, unless we had other plans for the campaign emails besides just deleting them. 😄

Reply:
Other plans? Well I was going to add support to Tweet out emails before being deleted, but fair enough I'll rename ;-)